Review and move SP how-to - Set up Data Validation

Yiyun333 opened the pull request:
Move how-to to refguide

Yiyun333 committed:
Review and move SP how-to - Set up Data Validation

Yiyun333 committed:
Update cross-references

Yiyun333 committed:
Move attachments

Yiyun333 committed:
Language review

Yiyun333 committed:
More language and style review

Yiyun333 committed:
Add cross-references for validation assist

Yiyun333 committed:
More rewrite and review

Yiyun333 committed:
Update screenshots and revise content

Yiyun333 committed:
Update screenshots and content

Yiyun333 committed:
small fix

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+## 1 Introduction
+
+This how-to explains how you can set up data validation with Mendix. Before you can start with this how-to, you need to set up a basic data structure, otherwise there is no data to validate. If you do not know how to set up a basic data structure, you can take a look at [Creating a Basic Data Layer](/refguide/create-a-basic-data-layer/).
```
```suggestion
This document explains how you can set up data validation with Mendix. Make sure you have set up a basic data structure, otherwise there is no data to validate. For more information on how to set up a basic data structure, see [Creating a Basic Data Layer](/refguide/create-a-basic-data-layer/).
```

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+{{% alert color="info" %}}
+Validation rules can only be applied on entities that are [persistable](/refguide/persistability/).
+{{% /alert %}}
+
```
Please add sentence before the list. For example, "follow the steps below".

Yiyun333 replied:
The introcutory sentence is added, see https://github.com/mendix/docs/pull/5764/commits/992a586a12f8d268b4511e242191d8a576edcbb0.

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+## 3 Required Validation on Inputs, Reference Selectors, and Drop-downs
+
+The page builder of Studio Pro allows you to configure which inputs are mandatory and what message is shown to the end-user if the input is empty. If you do not have a detail page yet, take a look at [How to create Your First Two Overview and Detail Pages](/howto/front-end/create-your-first-two-overview-and-detail-pages/).
```
What is a "page builder"? A "page editor"?
"The page builder of Studio Pro allows you to configure which inputs are mandatory and what message is shown to the end-user if the input is empty." - this sentence is a bit hard to read, consider rewriting it? For example, "allows you to configure mandatory inputs and messages shown for end-users if these inputs are empty".

Yiyun333 replied:
Yes, it should be a "page editor". The sentence is also rephrased: https://github.com/mendix/docs/pull/5764/commits/992a586a12f8d268b4511e242191d8a576edcbb0.

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+The page builder of Studio Pro allows you to configure which inputs are mandatory and what message is shown to the end-user if the input is empty. If you do not have a detail page yet, take a look at [How to create Your First Two Overview and Detail Pages](/howto/front-end/create-your-first-two-overview-and-detail-pages/).
+
+1. Open a **detail page** and double-click an input widget to open its properties.
```
Please add an introductory sentence before the list.

Yiyun333 replied:
Also done (in the same commit above).

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+{{< figure src="/attachments/refguide/modeling/domain-model/setting-up-data-validation/input-widget-validation.png" width="500px">}}
+
+{{% alert color="info" %}}
+**Required** validation on **Input elements** in a page editor always checks before the validation rule set on entity level. This means that if you set both as **Required** and have different error messages. The page error message is shown to the end-user.  
```
"before the validation rule **is** set"? Something is missing there.

Yiyun333 replied:
The updated note is this:
**Required** validation on **Input elements** in a page editor is always checked before the validation rule set on entity level is. This means that if you set both as **Required** and have different error messages, the page error message is shown to the end-user.

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+## 4 Advanced Data Validation with the 'Before Commit' Event
+
+Validation rules are great for simple validations, but Mendix also offers ways to handle more complex validations. The domain model allows you to define event handlers on entity level. The 'Before Commit' and 'After Commit' events are triggered when an object is committed to the database. The 'After Commit' is most commonly used to calculate values of denormalized data. With the 'Before Commit' e ...
```
Single quotation marks is not our style guide, perhaps **After Commit** and **Before Commit** should be bolded if we mean a UI element?

Yiyun333 replied:
Yeah, these UI elements are bolded now. It's a bit tricky here, because it's a combination of two seperate UI elements. But still better to bold them than use single quotation marks.

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+    {{< figure src="/attachments/refguide/modeling/domain-model/setting-up-data-validation/microflow-1.png" >}}
+
+As long as this microflow returns a Boolean value, you are free to add any logic to determine if the data is valid or not. If the microflow returns `false`, the commit is cancelled. Returning `true` commits the object. 
```
```suggestion
As long as this microflow returns a Boolean value, you are free to add any logic to determine whether the data is valid or not. If the microflow returns `false`, the commit is cancelled. Returning `true` commits the object. 
```

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+For more information on working with microflows, see [Microflows](/refguide/microflows/).
+
+## 5 Advanced Validation with a Custom 'Save' Button{#custom-validation-save-button}
```
```suggestion
## 5 Advanced Validation with a Custom 'Save' Button {#custom-validation-save-button}
```

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+## 5 Advanced Validation with a Custom 'Save' Button{#custom-validation-save-button}
+
+Validating user input can also be achieved by overriding the default **Save** button on a detail page. If you do not have a detail page yet, take a look at [How to create Your First Two Overview and Detail Pages](/howto/front-end/create-your-first-two-overview-and-detail-pages/) and create one.
```
"If you do not have a detail page yet, take a look at", not really our style guide.

Yiyun333 replied:
Changed the sentence to "For more information on how to create a detail page, see [How to Create Your First Two Overview and Detail Pages](/howto/front-end/create-your-first-two-overview-and-detail-pages/)."

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+There are two ways to configure a custom **Save** button for data validation. 
+
+You can do this with the help of [Validation Assist](/refguide/validation-assist/). With Validation Assist, you can build validation microflows in a more automatic way using pre-built expressions. For more information on how to use the Validation Assist, see the [Using Validation Assist to Build Data Validations](/refguide/validation-assist/#3-using-validation-assist-to-build-data-validations) se ...
```
```suggestion
You can do this with the help of [Validation Assist](/refguide/validation-assist/). With Validation Assist, you can build validation microflows automatically by using pre-built expressions. For more information on how to use the Validation Assist, see the [Using Validation Assist to Build Data Validations](/refguide/validation-assist/#3-using-validation-assist-to-build-data-validations) section in *Validation Assist*. 
```

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+    {{< figure src="/attachments/refguide/modeling/domain-model/setting-up-data-validation/microflow-4.png" width="500px" >}}
+
+11. Draw an additional line from the decision and add another end event.
```
We call "an additional line" a flow.

Yiyun333 replied:
Changed for all the instances.

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+
+11. Draw an additional line from the decision and add another end event.
+12. Right-click the line between the decision and the end event and select **false** as the condition value.
+13. Insert a **Validation feedback** activity between the decision and the new end event. In the activity's properties editor, configure the activity in the following way:
```
There is no "properties editor" term.

Yiyun333 replied:
Yeah, it should be "properties dialog box".

MariaShaposhnikova commented on content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md:
```diff
@@ -0,0 +1,147 @@
+## 7 Read More
+
+* [Validation Rules](/refguide/validation-rules/)
+* [Validation Section](/refguide/common-widget-properties/#validation)
```
We do not cross-reference section like this, please follow our style guide.

Yiyun333 replied:
I deleted all the repetitive cross-references in this section.

MariaShaposhnikova reviewed (commented):
Hi Yiyun,
Overall it looks good, some comments are about the Style guide. Perhaps the ones that require the most attention are terminology-related, this document uses several terms that we do not use in the documentation. Terms for microflows are mostly mention here: https://docs.mendix.com/refguide/microflows/. More general terms for SP are mention in this doc: https://docs.mendix.com/refguide/studio-pro-overview/

Yiyun333 commented:
> Hi Yiyun, Overall it looks good, some comments are about the Style guide. Perhaps the ones that require the most attention are terminology-related, this document uses several terms that we do not use in the documentation. Terms for microflows are mostly mention here: https://docs.mendix.com/refguide/microflows/. More general terms for SP are mention in this doc: https://docs.mendix.com/refguide/studio-pro-overview/

Hi @MariaShaposhnikova,
Thanks for the review and the clarification on terminologies! I'll go through the comments and make changes accordingly.

Yiyun333 committed:
Update content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md

Co-authored-by: Maria Shaposhnikova <38249778+MariaShaposhnikova@users.noreply.github.com>

Yiyun333 committed:
Update content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md

Co-authored-by: Maria Shaposhnikova <38249778+MariaShaposhnikova@users.noreply.github.com>

Yiyun333 committed:
Update content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md

Co-authored-by: Maria Shaposhnikova <38249778+MariaShaposhnikova@users.noreply.github.com>

Yiyun333 committed:
Update content/en/docs/refguide/modeling/domain-model/setting-up-data-validation.md

Co-authored-by: Maria Shaposhnikova <38249778+MariaShaposhnikova@users.noreply.github.com>

Yiyun333 committed:
Adress the feedback from Maria

Yiyun333 committed:
small fixes

Yiyun333 commented:
> Hi Yiyun, Overall it looks good, some comments are about the Style guide. Perhaps the ones that require the most attention are terminology-related, this document uses several terms that we do not use in the documentation. Terms for microflows are mostly mention here: https://docs.mendix.com/refguide/microflows/. More general terms for SP are mention in this doc: https://docs.mendix.com/refguide/studio-pro-overview/

Hi @MariaShaposhnikova,
I addressed all the feedback just now. I also marked most of the feedback as resolved, and left two that might need your double confirmation. Please let me know what you think, and if you agree, I'll merge the PR. Thanks!

Yiyun333 committed:
Revise the note

MariaShaposhnikova commented:
@Yiyun333 , thanks for implementing comments and addressing feedback, the PR looks good, please merge.
